### PR TITLE
Robust audio - Part 1

### PR DESCRIPTION
**Pull request by JustinTrotter**
### Purpose
 <br>

Set up foundation for making audio more robust.
Goals is complete task: https://github.com/unitystation/unitystation/issues/6787
### Notes:

Added proper audio mixer for master, music, soundfx, ambience, and tts
Updated sound options to audio options to include volume controls for each listed above.
Fixed admin global sounds.
Created admin global music.

### Changelog:

CL: [New] Separate audio controls for sound, music, and tts
CL: [Fix] Fixed admin global sounds
CL: [New] Added admin global music

### CODE REVIEW

**Comment by NoooneyDude** [comment]
Looking forward to this.

**Comment by corp-0** [comment]
looks good so far, I will test tomorrow if I can

File: UnityProject/Assets/Scripts/Managers/AudioManager.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            get
+            {
+                if (audioManager == null)
+                {
+                    audioManager = FindObjectOfType<AudioManager>();
+                }
+
+                return audioManager;
+            }
+        }
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Managers/AudioManager.cs]:
```suggestion
    public class AudioManager : SingletonManager
    {
       
```

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Managers/AudioManager.cs]: You might need to do ` public class AudioManager : SingletonManager<AudioManager>` , cant remember the exact format

**Comment by JustinTrotter** [comment]
Updated with new features. I need to validate some things. DNM for now.

File: UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs [hunk 2]
```diff
@@ -93,13 +93,13 @@ public void CmdToggleOOCMute(NetworkConnectionToClient sender = null)
 		#region EventsPage
 
 		[Command(requiresAuthority = false)]
-		public void CmdTriggerGameEvent(int eventIndex, bool isFake, bool announceEvent,
+		public void CmdTriggerGameEvent(int eventIndex, int musicIndex, bool isFake, bool announceEvent,
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs]: Not a fan of this new parameter, the global sound player just needs fixing, you dont need to make an event to play music. 

The sound player needs to be refactored for addressable audio. #6235

Comment (JustinTrotter) [reply UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs]: Part of the requirements was to have admin triggered songs and to have music played during in game events.

Should I create a new admin modal to handle the global sound player?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs]: > Part of the requirements was to have admin triggered songs and to have music played during in game events.

The global sound player (currently broken) is for admin trigger sounds/music. 

The in game events requirement just seems to mean when music is triggered by an event. So it's build a system for playing music which has priorities and then if an event starts music(which'll be in the event code not admin chosen) then it should stop the current music if its a lower priority and play.

> Should I create a new admin modal to handle the global sound player?

Not sure what you mean by this. The global sound player is the button on the top of the screen when you are an admin, think the button just says `Global Sounds`.

Comment (JustinTrotter) [reply UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs]: > The in game events requirement just seems to mean when music is triggered by an event. So it's build a system for playing music which has priorities and then if an event starts music(which'll be in the event code not admin chosen) then it should stop the current music if its a lower priority and play.

Okay. Sounds good.

> The global sound player is the button on the top of the screen when you are an admin, think the button just says `Global Sounds`.

I didn’t realize this existed. Thanks.

**Comment by JustinTrotter** [comment]
Networked admin sounds are now working! It uses addressable audio too. Next step is to add networked admin music and make sure it plays on the music audio mixer. Then I'll look into having the music fade in and out when transitioning from one song to another. 

~~I still need to undo the InGameEventsManager changes.~~
I also need to refactor AudioManager to use SingletonManager. 

At that point I'd consider part 1 to be ready for review and possible merge.

**Comment by corp-0** [comment]
adding on hold label until it is ready to review again. Please let us know then.

**Comment by JustinTrotter** [comment]
Admin music added. Still need to validate that it works over a network.

Still need to refactor and clean up some things.

**Review by ThatDan123 — APPROVED** [review]
Can't see any issues with the code, just needs testing.

**Review by corp-0 — APPROVED** [review]
The conceptual difference between the word audio and sound isn't immediately clear for me but the code looks good nonetheless

**Comment by JustinTrotter** [comment]
> The conceptual difference between the word audio and sound isn't immediately clear for me but the code looks good nonetheless

The way I see it is that sound and music is a subset of audio. Later on in other parts for this epic I’d like to move some of the functionality that SoundManager has into AudioManager to make it more generalized.

**Pull request merged**